packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -526,13 +525,6 @@ function performRecordIdentifierUpdate(
       }
     }
 
-    // TODO consider just ignoring here to allow flexible polymorphic support
-    if (type && type !== identifier.type) {
-      throw new Error(
-        `The 'type' for a RecordIdentifier cannot be updated once it has been set. Attempted to set type for '${wrapper}' to '${type}'.`
-      );
-    }
-
```

Comment:
Thanks @ryandoherty!  Do you have a test addition we can verify this results in added behaviour?

Reply:
I do not, I can give it a whirl. First time working on anything Ember, happy to take pointers to where/how I should do this. Thanks!

Reply:
I'd note that we do have tests for single-table polymorphism support in place already. We might be able to improve our default implementation but the intent with identifiers is that you'll configure the cache in your own app to handle edge cases around one record having multiple potential identities.